sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1534,15 +1534,15 @@ class Dataset[T] private[sql](
    * Returns a new Dataset by sampling a fraction of rows.
    *
    * @param withReplacement Sample with replacement or not.
-   * @param fraction Fraction of rows to generate.
+   * @param fraction Fraction of rows to generate and the range is 0.0 <= `fraction` <= 1.0.
    * @param seed Seed for sampling.
    *
    * @group typedrel
    * @since 1.6.0
    */
   def sample(withReplacement: Boolean, fraction: Double, seed: Long): Dataset[T] = {
-    require(fraction >= 0,
-      s"Fraction must be nonnegative, but got ${fraction}")
+    require(fraction >= 0 && fraction <= 1.0,
+      s"Fraction range must be 0.0 <= `fraction` <= 1.0, but got ${fraction}")
```

Comment:
Hi @maropu, I just wonder if this fix is needed though just to be consistent whether `withRelacement` is `true` or not.